dropgz/cmd/payload.go

```diff
@@ -32,7 +38,7 @@ func checksum(srcs, dests []string) error {
 	if len(srcs) != len(dests) {
 		return errors.Wrapf(embed.ErrArgsMismatched, "%d and %d", len(srcs), len(dests))
 	}
-	rc, err := embed.Extract("sum.txt")
+	rc, err := embed.Extract("sum.txt", compression)
```

Comment:
Were there issues with compression previously? Why are we adding an option to disable compression (since previously my understanding is that it was always enabled?)?

Reply:
Is it possible to test that checksum etc. is working when we don't have compression? Has it been tested manually once maybe?

Reply:
I have verified that everything still works as expected.
As for why - when I created dropgz it was to ship a bunch of blobs in a single executable that knew how to self-extract those blobs. I made it expect and require that the blobs be pre-gzipped because my use case was to put that dropgz executable in a container and use it to get CNI bins on to Nodes, and I expected that making the executable as small as possible would mean making the image as small as possible and would result in the minimal image pull time and fastest CNI ready. 
This may not actually be right...
Containers are stored and transmitted compressed (as gzip by default) and the double compression was not actually reducing the final transmitted image size. Instead, it was adding compute time to decompress twice.

Reply:
Also, we try to pre-cache images in the Node VHD, so transmit time on our happy path is 0. It's a waste of time to have additional decompression when the image is already on-disk.